src/main/scala/com/intenthq/action_processor/integrations/aggregations/Aggregate.scala

```diff
@@ -0,0 +1,62 @@
+package com.intenthq.action_processor.integrations.aggregations
+
+import java.util.concurrent.ConcurrentMap
+
+import cats.effect.{IO, Resource}
+import com.intenthq.action_processor.integrations.config.MapDbSettings
+import com.intenthq.action_processor.integrations.feeds.FeedContext
+import com.intenthq.action_processor.integrations.repositories.MapDBRepository
+import org.mapdb.elsa.{ElsaMaker, ElsaSerializer}
+import org.mapdb.serializer.GroupSerializerObjectArray
+import org.mapdb.{DataInput2, DataOutput2, HTreeMap, Serializer}
+
+import scala.jdk.CollectionConverters._
+
+object Aggregate {
+
+  def noop[I]: fs2.Pipe[IO, I, (I, Long)] = _.map(_ -> 1L)
+
+  private def loadAggRepository[K](mapDbSettings: MapDbSettings): Resource[IO, HTreeMap[K, Long]] = {
+
+    val serializer: Serializer[K] = new GroupSerializerObjectArray[K] {
+      val elsaSerializer: ElsaSerializer = new ElsaMaker().make
+      override def deserialize(input: DataInput2, available: Int): K = elsaSerializer.deserialize[K](input)
+      override def serialize(out: DataOutput2, value: K): Unit = elsaSerializer.serialize(out, value)
+    }
+
+    MapDBRepository
+      .load(mapDbSettings)
+      .map(db =>
+        db.hashMap("stuff", serializer, Serializer.LONG)
```

Comment:
I think this should work better instead of the later `.asInstanceOf[HTreeMap[K, Long]]`
```
db.hashMap("stuff", serializer, Serializer.LONG.asInstanceOf[scala.Long])
```
https://github.com/Karasiq/mapdbutils/blob/d2e6529b66a5fa4c7045473f26c28dc004011887/src/main/scala/com/karasiq/mapdb/serialization/PredefinedSerializers.scala#L18

Reply:
Really good catch! 👍🏼